org.eclipse.winery.topologymodeler.ui/src/app/canvas/canvas.component.ts

```diff
@@ -66,30 +67,30 @@ export class CanvasComponent implements OnInit, OnDestroy, AfterViewInit, DoChec
     @ViewChild('requirementsModal') requirementsModal: ModalDirective;
     @ViewChild('importTopologyModal') importTopologyModal: ModalDirective;
     @Input() entityTypes: EntityTypesModel;
-    @Input() relationshipTypes: Array<any> = [];
+    @Input() relationshipTypes: Array<EntityType> = [];
     @Input() diffMode = false;
 
+    readonly draggingThreshold = 300;
+    readonly newNodePositionOffsetX = 108;
+    readonly newNodePositionOffsetY = 30;
+
     allNodeTemplates: Array<TNodeTemplate> = [];
     allRelationshipTemplates: Array<TRelationshipTemplate> = [];
     navbarButtonsState: ButtonsStateModel;
     selectedNodes: Array<TNodeTemplate> = [];
     // current data emitted from a node
     currentModalData: any;
     dragSourceActive = false;
-    currentType: string;
+    selectedRelationshipType: string;
```

Comment:
Why is that still a string? Shouldn't this also be of type `EntityType`?

Reply:
no, only the relationship name is passed because there is no access to the whole relationship EntityType object since the mousedown event handler is not in the ngFor loop anymore (otherwise no relationship can be dragged)